### ISSUE TITLE
Feature request: Server setting: Unfreeze teleport

### ISSUE THREAD

**Issue by BannZay** [ddnet/ddnet#2385]
This is common practice to place unfreeze tile on top of teleto entity. Can we have a server setting to unfreeze any teleported tee ?
1.It will reduce amount of noize on the maps, 2. Mappers will be happy to not manage game layer everytime when you tweak teleto location

**Comment by heinrich5991** [comment]
Server setting is not a good idea here IMO as it makes maps less predictable. I could imagine another tile, but then this is just basically the same as mapping unfreeze and teleto.

Perhaps one could add a setting to the editor to automatically map unfreeze when you add a teleto and remove it when you remove the teleto.

**Comment by BannZay** [comment]
> makes maps less predictable
does it? I would say opposite if some teleports dont unfreeze you but most do - that is a thing that makes map less predictable

And I guess that these tiles(unfreeze on top of tele) is just a straightforward hack to make things work properly i.e. when we dont have such setting but we want to simulate this behavior

**Comment by heinrich5991** [comment]
It makes maps less predictable when you play them with entities. Some would unfreeze on teleporters without unfreeze, some wouldn't.

**Comment by BannZay** [comment]
oh. I wanted a setting to make ALL teleports unfreeze u

**Comment by BannZay** [comment]
I believe this problem not worth keeping random unfreezes floating around just to make teleports be unfreezable

**Comment by C0D3D3V** [comment]
I think this could be implemented using a combination of teleport to and unfreeze, like it is done for the new hook through.

**Comment by C0D3D3V** [comment]
Ah I see your argument was:

> putting unfreeze on top of teleto makes this place behave as unfreeze which might be unwanted by mapper, as well mapper has to carry about putting unfreezes on each teleto. As a mapper I would like to have such type of teles

I only thought I convert this request into a request for a new tile since your config request was rejected. But I guess it should not be a combination of to tele + unfreeze, but the tele to needs to be a new kind of tele to, because it should not unfreeze you if you get contact with it without teleporting.

**Comment by heinrich5991** [comment]
Perhaps it could be an editor setting instead? That you always map an unfreeze together with a teleporter instead?

We already have the functionality, adding a tile doesn't seem to be useful.

**Comment by C0D3D3V** [comment]
The idea is to only unfreeze "teleported" players, not players that touch the tele to, so it would need to be an option for the tele to tile. Just adding a combination of unfreeze + tele to does not solve the problem

**Comment by heinrich5991** [comment]
I see. Does that work if you put the unfreeze on the telefrom?

I don't think we should add a tile for every combination of features. Fewer tiles is better for players and also for code maintenance as there are less bugs with less code.

**Comment by C0D3D3V** [comment]
Yes, I also have seen the tele from combination with unfreeze on some maps. This should produce the outcome we want here. 
So we could add combination tiles for the 4 tele from tiles, that add unfreeze to them.

**Comment by BannZay** [comment]
> Yes, I also have seen the tele from combination with unfreeze on some maps. 

It will not, it is not reliable. It does not unfreeze sometimes but just teleports.

**Comment by BannZay** [comment]
Btw there was a reason to make hookthrough as a combined tile but not a single one. This case is another story.

**Comment by BannZay** [comment]
@heinrich5991 here is the deal, we can render fake unfreeze tiles on top of teleports in entities mode if the setting is set, in case if you do really conserned with "predictability" problem.

**Comment by heinrich5991** [comment]
> And this solution still requires two actions from mapper instead of one which is not good for a widely used feature.

I'm not opposed to fixing this.

> Or about other map settings?

Yes. Unpredictability is bad for everyone. It's good if you can look at a map and have an idea of what it does.

**Comment by C0D3D3V** [comment]
From a player point of view, a lot of ddnet maps are not predictable anyway. Often unfreeze is not added to the tile layer. Somtimes you can not tell the difrence between the red and blue tele without looking at entities. You can not even tell if unfreeze is set on tele from, because the mapper will mostly set in tile layer only some darker tile for teleport, that does not tell you anything about the tile, and in entities you can mostly can not tell the difrence because it looks nearly the same as without freeze behind the tele. Often you can not tell if there are diffrent teleports next to each other, without entities. Somtimes checkoints are in tile layer only small, but in entities the checkpoint goes up to the top of the room. 


Entities has become a feature to predict bad mapped maps.
